BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1874 @@
++    }
++    // Waterfall end
+ }
+diff --git a/api/src/main/java/net/md_5/bungee/api/score/Team.java b/api/src/main/java/net/md_5/bungee/api/score/Team.java
+index 849ba1cf..db3f95b0 100644
+--- a/api/src/main/java/net/md_5/bungee/api/score/Team.java
++++ b/api/src/main/java/net/md_5/bungee/api/score/Team.java
+@@ -6,6 +6,10 @@ import java.util.HashSet;
+ import java.util.Set;
+ import lombok.Data;
+ import lombok.NonNull;
++// Waterfall start - we're going on an adventure!
++import net.kyori.adventure.text.Component;
++import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
++// Waterfall end
+ 
+ @Data
+ public class Team
+@@ -13,9 +17,14 @@ public class Team
+ 
+     @NonNull
+     private final String name;
+-    private String displayName;
+-    private String prefix;
+-    private String suffix;
++    // Waterfall start - we're going on an adventure!
++//    private String displayName;
++    private Component displayName;
++//    private String prefix;
++    private Component prefix;
++//    private String suffix;
++    private Component suffix;
++    // Waterfall end
+     private byte friendlyFire;
+     private String nameTagVisibility;
+     private String collisionRule;
+@@ -36,4 +45,62 @@ public class Team
+     {
+         players.remove( name );
+     }
++
++    // Waterfall start - add previously Lombok-generated getters/setters - we're going on an adventure!
++    @Deprecated
++    public String getDisplayName() {
++        return LegacyComponentSerializer.legacySection().serialize(displayName);
++    }
++
++    @Deprecated
++    public void setDisplayName(final String displayName) {
++        this.displayName = LegacyComponentSerializer.legacySection().deserialize(displayName);
++    }
++
++    @Deprecated
++    public String getPrefix() {
++        return LegacyComponentSerializer.legacySection().serialize(prefix);
++    }
++
++    @Deprecated
++    public void setPrefix(final String prefix) {
++        this.prefix = LegacyComponentSerializer.legacySection().deserialize(prefix);
++    }
++
++    @Deprecated
++    public String getSuffix() {
++        return LegacyComponentSerializer.legacySection().serialize(suffix);
++    }
++
++    @Deprecated
++    public void setSuffix(final String suffix) {
++        this.suffix = LegacyComponentSerializer.legacySection().deserialize(suffix);
++    }
++    // Waterfall end
++
++    // Waterfall start - we're going on an adventure!
++    public Component displayName() {
++        return displayName;
++    }
++
++    public void displayName(final Component displayName) {
++        this.displayName = displayName;
++    }
++
++    public Component prefix() {
++        return prefix;
++    }
++
++    public void prefix(final Component prefix) {
++        this.prefix = prefix;
++    }
++
++    public Component suffix() {
++        return suffix;
++    }
++
++    public void suffix(final Component suffix) {
++        this.suffix = suffix;
++    }
++    // Waterfall end
```

Comment:
Won't you have duplicate functions here when lombok generates it's functions with `get`/`set` prefixes?

Reply:
I know that the getters definitely will not, as we have custom getters with the same name and so Lombok won’t generate the getters as that would produce a compiler error. I am unsure about the setters actually though, perhaps I should mark these as access level none

Reply:
Also, just found out that Lombok is even stupider and just searches for a method with the name that matches the getter/setter it tries to generate and just refuses to generate said getter/setter, so these are fine here.